jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/jsonb/JsonbDiagnosticsCollectorTest.java

```diff
@@ -86,7 +88,22 @@ public void JsonbTransientNotMutuallyExclusive() throws Exception {
         Diagnostic d1 = d(22, 19, 23,
                 "@JsonbTransient must be mutually exclusive with all other JSON Binding defined annotations.",
                 DiagnosticSeverity.Error, "jakarta-jsonb", "NonmutualJsonbTransientAnnotation");
+        d1.setData(new Gson().toJsonTree(Arrays.asList("JsonbProperty", "JsonbTransient")));
+        
+        Diagnostic d2 = d(27, 19, 23,
+                "@JsonbTransient must be mutually exclusive with all other JSON Binding defined annotations.",
+                DiagnosticSeverity.Error, "jakarta-jsonb", "NonmutualJsonbTransientAnnotation");
+        d2.setData(new Gson().toJsonTree(Arrays.asList("JsonbProperty", "JsonbDateFormat", "JsonbTransient")));
```

Comment:
Is there a quick-fix we should test for associated with d2?

Reply:
It seems that it is not possible to test for diagnostics where the quick fix involves removing more than one annotation at a time.
For example, the following is resulting in an error
```java
        JakartaJavaCodeActionParams codeActionParams2 = createCodeActionParams(uri, d2);
        TextEdit te3 = te(26, 4, 27, 4, "");
        TextEdit te4 = te(24, 4, 26, 4, "");
        CodeAction ca3 = ca(uri, "Remove @JsonbTransient", d2, te3);
        CodeAction ca4 = ca(uri, "Remove @JsonbProperty, @JsonbDateFormat", d2, te4);
        assertJavaCodeAction(codeActionParams2, utils, ca3, ca4);
```

This leads to a NullPointerException that originates from line 65 of the following file https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JakartaForJavaAssert.java#L65

The reason for this null pointer exception is that, the `parentType` returned in the testing environment is null, therefore the following check `if (parentType == null)` on line 57 on `JsonbTransientAnnotationQuickfix.java` fails and `null` is returned.

All other diagnostics that use this type of logic (removing more than one annotation at once ignore this test).
Any ideas on how to deal with this? Or should I create a new issue for this? I don't seem to quite get why the 'parentType' returned on line 54 is null _only_ when the quick fix suggests removing more than one annotation at a time.

Reply:
I am fine to remove the null check